src/app/qgsfieldsproperties.cpp

```diff
@@ -305,7 +305,14 @@ void QgsFieldsProperties::setRow( int row, int idx, const QgsField& field )
     mFieldsList->setItem( row, attrCommentCol, new QTableWidgetItem( field.comment() ) );
   }
 
-  for ( int i = 0; i < attrCommentCol; i++ )
+  QList<int> notEditableCols = QList<int>()
+                               << attrIdCol
+                               << attrNameCol
+                               << attrTypeCol
+                               << attrTypeNameCol
+                               << attrLengthCol
+                               << attrPrecCol;
+  Q_FOREACH ( int i, notEditableCols )
```

Comment:
Actually it's other way around. 
The old code was considering the order of columns fixed. The new one considers individual columns which should be set as read only.